charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml

```diff
@@ -88,6 +88,9 @@ livenessProbe:
 ## @param podAnnotations [object] Pod annotations for Spire OIDC discovery provider
 podAnnotations: {}
 
+## @param tlsSecret Provide your own certificate/key via tls style Kubernetes Secret
+tlsSecret: ""
```

Comment:
I think it would be good to link this, cert-manager, and spiffe-helper as the source of certs for this. Maybe have all 3 under a sub section?

Reply:
hmm... what would that look like?

maybe something like:
```
tls:
  mode: [spiffe or cert-manager or external or none]
  externalSecret: "" #used if mode==external
  cert-manager: ...... stuff from below?
```

where mode=none replaces the insecureScheme.enabled flag?

Reply:
I think we can do it without the enum:
```
tls:
  spire:
    enabled: true
    ....

  externalSecret:
    enabled: false
    secretName: my-secret

  cert-manager:
     enabled: true
```
And print an error if more than one is set. What do you think?

Reply:
Updated.

Reply:
Thanks @kfox1111 for refactoring. I think its easier to follow now.